js/src/components/tree-select-control/index.js

```diff
@@ -92,14 +110,16 @@ const TreeSelectControl = ( {
 	const [ treeVisible, setTreeVisible ] = useState( false );
 	const [ nodesExpanded, setNodesExpanded ] = useState( [] );
 	const [ inputControlValue, setInputControlValue ] = useState( '' );
-	const [ filteredOptions, setFilteredOptions ] = useState( [] );
 
 	const dropdownRef = useRef();
 	const onDropdownVisibilityChangeRef = useRef();
 	onDropdownVisibilityChangeRef.current = onDropdownVisibilityChange;
 
 	// We will save in a REF previous search filter queries to avoid re-query the tree and save performance
 	const filteredOptionsCache = useRef( {} );
+	const cacheStatesRef = useRef( {} );
+	cacheStatesRef.current.expandedValues = nodesExpanded;
+	cacheStatesRef.current.selectedValues = value;
```

Comment:
💅  WDTY of saving all the caches in

`cacheRef.current.filteredOptions`
`cacheRef.current.expandedValues `
`cacheRef.current.selectedValues `

Reply:
Changed in https://github.com/woocommerce/google-listings-and-ads/pull/1522/commits/e2a6a827289312f796e202e68b628ed34f0485d4.